aws/resource_aws_kinesis_firehose_delivery_stream.go

```diff
@@ -2501,17 +2581,22 @@ func waitForKinesisFirehoseDeliveryStreamSSEDisabled(conn *firehose.Firehose, de
 }
 
 func isKinesisFirehoseDeliveryStreamOptionDisabled(v interface{}) bool {
-	options := v.([]interface{})
-	if len(options) == 0 || options[0] == nil {
-		return true
-	}
-	m := options[0].(map[string]interface{})
+	optionMap := getOptionsOfConfigurationUnit(v)
 
 	var enabled bool
 
-	if v, ok := m["enabled"]; ok {
+	if v, ok := optionMap["enabled"]; ok {
 		enabled = v.(bool)
 	}
 
 	return !enabled
 }
+
+// helper to get options of a unit
+func getOptionsOfConfigurationUnit(v interface{}) map[string]interface{} {
```

Comment:
I understand your point, but I'm not quite sure, how to resolve this problem. The easiest thing would be to undo the changes and copy the code to the functions, if required. But this is of course not very beautiful to read and handle, since it would be repeating code then.
Since the issue is just for adding CMK support and not changing the structure of the file to save some code, I wouldn't recommend to make this change even more difficult to understand (like already refactoring this is a change, which is not really understandable when reading the issue's title).

Do you have any idea how to change this or should this be part of another issue?